components-java-sdk/src/test/java/io/dapr/components/wrappers/StateStoreGrpcComponentWrapperTest.java

```diff
@@ -0,0 +1,180 @@
+package io.dapr.components.wrappers;
+
+import dapr.proto.components.v1.State;
+import dapr.proto.components.v1.StateStoreGrpc;
+import io.dapr.components.domain.state.StateStore;
+import io.dapr.v1.ComponentProtos;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import io.grpc.testing.GrpcCleanupRule;
+import org.junit.Rule;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static io.dapr.components.TestUtils.assertThrowsGrpcRuntimeException;
+import static io.dapr.components.TestUtils.monoError;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.anyMap;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class StateStoreGrpcComponentWrapperTest {
+
+  /**
+   * This rule manages automatic graceful shutdown for the registered servers and channels at the
+   * end of test.
+   */
+  @Rule
+  public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  private StateStore mockComponent;
+  private StateStoreGrpc.StateStoreBlockingStub client;
+
+  @BeforeEach
+  void setUp() throws IOException {
+    mockComponent = mock();
+
+    // Generate a unique in-process server name.
+    final String serverName = InProcessServerBuilder.generateName();
+
+    // Create a server, add service, start, and register for automatic graceful shutdown.
+    grpcCleanup.register(InProcessServerBuilder
+        .forName(serverName)
+        .directExecutor()
+        .addService(new StateStoreGrpcComponentWrapper(mockComponent))
+        .build()
+        .start()
+    );
+
+    client = StateStoreGrpc.newBlockingStub(
+        // Create a client channel and register for automatic graceful shutdown.
+        grpcCleanup.register(InProcessChannelBuilder.forName(serverName).directExecutor().build()));
+  }
+
+  @AfterEach
+  void tearDown() {
+  }
+
+  //
+  // Tests for common component aspects
+  //
+
+  @Test
+  void initHappyCase() {
```

Comment:
I wish there were a way to reuse tests for init/ping/features across component types, but the generated types don't make that easy to setup.

Reply:
I did managed to get something working around that on dcc03bc6756a629182fa9f8854c740bf9618c86a. Let me know what you think.